paimon-core/src/test/java/org/apache/paimon/catalog/CatalogTestBase.java

```diff
@@ -603,11 +603,11 @@ public void testAlterTableUpdateColumnType() throws Exception {
                                         identifier,
                                         Lists.newArrayList(
                                                 SchemaChange.updateColumnType(
-                                                        "col1", DataTypes.STRING())),
+                                                        "col1", DataTypes.DATE())),
```

Comment:
Why changing this?

Reply:
This test case is Alter table update a column type throws Exception when column data type does not support cast, this PR is modified to support BIGINT to STRING, this case is correct, so I changed it to BIGINT to DATE.

Reply:
Each PR should be complete by its own. This PR is not for solving bugs, but for introducing a new feature. By introducing this new feature do you really want to create new bugs and just leave them behind? Please fix this.

Reply:
New support for type cast: String to Number, Boolean to Number or String, Timestamp to Number or String.